sklearnex/tests/_utils_spmd.py

```diff
@@ -0,0 +1,181 @@
+    """
+
+    # create sycl queue and gather communicator details
+    q = SyclQueue("gpu")
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    size = comm.Get_size()
+
+    # divide data across ranks and move to dpt tensor
+    data_rows = full_data.shape[0]
+    local_start = rank * data_rows // size
+    local_end = (1 + rank) * data_rows // size
+    local_data = full_data[local_start:local_end]
+
+    return local_data
+
+
+def _generate_regression_data(n_samples, n_features, random_state=42):
+    # Generates regression data and divides between train and test
+    X, y = make_regression(
+        n_samples=n_samples, n_features=n_features, random_state=random_state
+    )
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=random_state)
+    return X_train, X_test, y_train, y_test
+
+
+def _generate_classification_data(n_samples, n_features, n_classes=2, random_state=42):
+    # Generates classification data and divides between train and test
+    X, y = make_classification(
+        n_samples=n_samples,
+        n_features=n_features,
+        n_classes=n_classes,
+        n_informative=int(0.5 * n_classes + 1),
+        random_state=random_state,
+    )
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=random_state)
+    return X_train, X_test, y_train, y_test
+
+
+def _generate_statistic_data(n_samples, n_features, random_state=42):
+    # Generates statistical data and divides between train and test
+    gen = np.random.default_rng(random_state)
+    data = gen.uniform(low=-0.3, high=+0.7, size=(n_samples, n_features))
+    return data
+
+
+def _generate_clustering_data(n_samples, n_features, centers=None, random_state=42):
+    # Generates clustering data and divides between train and test
+    X, y = make_blobs(
+        n_samples=n_samples,
+        centers=centers,
+        n_features=n_features,
+        random_state=random_state,
+    )
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=random_state)
+    return X_train, X_test, y_train, y_test
+
+
+def _spmd_assert_allclose(spmd_result, batch_result, **kwargs):
+    """Calls assert_allclose on spmd and batch results.
+
+    Called on each rank to compare the spmd result specific to that rank and
+    subset of batch result that corresponds to that rank.
+
+    Args:
+        spmd_result (numpy or dpctl array): The result for the subset of data on the rank the function is called from, computed by the spmd estimator
+        batch_result (numpy array): The result for all data, computed by the batch estimator
+
+    Raises:
+        AssertionError: If all results are adequately close.
+    """
+
+    # extract chunk from batch result to match with local spmd result
+    local_batch_result = _get_local_tensor(batch_result)
+
+    # convert to numpy if needed
+    if not isinstance(spmd_result, np.ndarray):
+        numpy_spmd_result = dpt.to_numpy(spmd_result)
```

Comment:
Better to use `_as_numpy` https://github.com/intel/scikit-learn-intelex/blob/04dbc5e1a412ac76f3578f76508df87e46487910/onedal/tests/utils/_dataframes_support.py#L115C5-L115C14

Reply:
Good catch, meant to remove this previously